pkg/telemetryquerylanguage/functions/tqlotel/func_replace_all_patterns.go

```diff
@@ -23,7 +23,12 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
 )
 
-func ReplaceAllPatterns(target tql.GetSetter, regexPattern string, replacement string) (tql.ExprFunc, error) {
+const (
+	modeKey   = "key"
+	modeValue = "value"
+)
+
+func ReplaceAllPatterns(target tql.GetSetter, mode string, regexPattern string, replacement string) (tql.ExprFunc, error) {
```

Comment:
Can you add validation that mode is exactly `key` or `value`?

Reply:
Done